mne/minimum_norm/psf_ctf.py

```diff
@@ -105,81 +105,44 @@ def point_spread_function(inverse_operator, forward, labels, method='dSPM',
 
     forward = convert_forward_solution(forward, force_fixed=False,
                                        surf_ori=True)
-    info = _prepare_info(inverse_operator)
-    leadfield = _pick_leadfield(forward['sol']['data'][:, 2::3], forward,
-                                info['ch_names'])
 
-    # will contain means of subleadfields for all labels
-    label_psf_summary = []
-    # if mode='svd', this will collect all SVD singular values for labels
-    label_singvals = []
+    info_inv = _prepare_info(inverse_operator)
 
-    # loop over labels
-    for ll in labels:
-        logger.info(ll)
-        if ll.hemi == 'rh':
-            # for RH labels, add number of LH vertices
-            offset = forward['src'][0]['vertno'].shape[0]
-            # remember whether we are in the LH or RH
-            this_hemi = 1
-        elif ll.hemi == 'lh':
-            offset = 0
-            this_hemi = 0
+    fwd_ch_names = forward['info']['ch_names']
+    inv_ch_names = info_inv['ch_names']
+    ch_names = [c for c in inv_ch_names if (c not in info_inv['bads']) ]
 
-        # get vertices on cortical surface inside label
-        idx = np.intersect1d(ll.vertices, forward['src'][this_hemi]['vertno'])
+    # select those bad channels that are actually used in current operator
+    # (otherwise could cause trouble later)
+    ch_bads = [c for c in info_inv['bads'] if (c in inv_ch_names)]
+    forward['info']['bads'] = ch_bads
+    inverse_operator['info']['bads'] = ch_bads
 
-        # get vertices in source space inside label
-        fwd_idx = np.searchsorted(forward['src'][this_hemi]['vertno'], idx)
+    # reduce forward to channels in invop
+    forward = pick_channels_forward(forward, ch_names)
 
-        # get sub-leadfield matrix for label vertices
-        sub_leadfield = leadfield[:, fwd_idx + offset]
+    leadfield = _pick_leadfield(forward['sol']['data'][:, 2::3], forward,
+                                                                   ch_names)
 
-        # compute summary data for labels
-        if mode == 'sum':  # sum across forward solutions in label
-            logger.info("Computing sums within labels")
-            this_label_psf_summary = sub_leadfield.sum(axis=1)[np.newaxis, :]
-        elif mode == 'mean':
-            logger.info("Computing means within labels")
-            this_label_psf_summary = sub_leadfield.mean(axis=1)[np.newaxis, :]
-        elif mode == 'svd':  # takes svd of forward solutions in label
-            logger.info("Computing SVD within labels, using %d component(s)"
-                        % n_svd_comp)
+    # average-referencing leadfield for EEG before SVD    
+    EEG_idx = [cc for cc in range(len(ch_names)) if ch_names[cc][:3]=='EEG']
```

Comment:
we hsould use 'pick_types' here, this is too brittle. Not every device will labels EEG channels like this.